tests/integration/test_self_monitoring.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+
+import pytest
+import requests
+from helpers import get_unit_address, mimir_endpoint_request, oci_image
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+MIMIR = "mimir"
+PROMETHEUS = "prometheus"
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_and_relate_charms(ops_test: OpsTest):
+    """Test that Mimir can be related with Prometheus over prometheus_scrape."""
+    # Build charm from local source folder
+    mimir_charm = await ops_test.build_charm(".")
```

Comment:
This could be time-memoized

1. Add timed memoizer to conftest ([ex.](https://github.com/canonical/grafana-k8s-operator/blob/aac2dcb01d0ad5c8003f024696e53d415039aeef/tests/integration/conftest.py#L74))
2. Then pass it as a fixture ([ex.](https://github.com/canonical/grafana-k8s-operator/blob/aac2dcb01d0ad5c8003f024696e53d415039aeef/tests/integration/test_get_password.py#L27))